### PR TITLE
VIH-10936 - Handle the callback event

### PR DESCRIPTION
**Pull request by marcogagliardi**
### Jira link
[VIH-10936](https://tools.hmcts.net/jira/browse/VIH-10936)

### Change description
Create new event handler, Create new eventType enum, Create UnitTest,  Added IEventHubClient

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: VideoWeb/VideoWeb.EventHub/Hub/IEventHubClient.cs [hunk 1]
```diff
@@ -17,6 +17,7 @@ public interface IEventHubClient
         Task EndpointStatusMessage(Guid endpointId, Guid conferenceId, EndpointState endpointState);
         Task ConferenceStatusMessage(Guid conferenceId, ConferenceStatus conferenceState);
         Task CountdownFinished(Guid conferenceId);
+        Task RecordingConnectionFailed(Guid conferenceId, Guid participantId);
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb.EventHub/Hub/IEventHubClient.cs:20]: If you're not going to invoke this, then it's not worth adding

File: VideoWeb/VideoWeb.UnitTests/EventHandlers/RecordingConnectionEventHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                EventType = EventType.RecordingConnectionFailed,
+                EventId = Guid.NewGuid().ToString(),
+                ParticipantId = participantId,
+                ConferenceId = conference.Id,
+                TimeStampUtc = DateTime.UtcNow
+            };
+
+            await _eventHandler.HandleAsync(callbackEvent);
+
+            // Verify messages sent to event hub clients
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb.UnitTests/EventHandlers/RecordingConnectionEventHandlerTests.cs:35]: I'm not sure what this test is doing. Do you want to add some asserts?

File: VideoWeb/VideoWeb.EventHub/Hub/IEventHubClient.cs (hunk 1 above)
Comment (shaed-parkar) [reply VideoWeb/VideoWeb.EventHub/Hub/IEventHubClient.cs:20]: Looking back at the ticket, the RecordingConnectionFailed should be published

**Review by shaed-parkar — COMMENTED** [review]
Looking good but the tests need to assert on behaviour, even if it's to assert the hubcontext is not invoked

File: VideoWeb/VideoWeb.EventHub/Hub/IEventHubClient.cs (hunk 1 above)
Comment (shaed-parkar) [reply VideoWeb/VideoWeb.EventHub/Hub/IEventHubClient.cs:20]: Keep it

File: VideoWeb/VideoWeb.UnitTests/EventHandlers/RecordingConnectionEventHandlerTests.cs (hunk 2 above)
Comment (marcogagliardi) [reply VideoWeb/VideoWeb.UnitTests/EventHandlers/RecordingConnectionEventHandlerTests.cs:37]: assert added

File: VideoWeb/VideoWeb.EventHub/Handlers/RecordingConnectionEventHandler.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        public override EventType EventType => EventType.RecordingConnectionFailed;
+
+        protected override async Task PublishStatusAsync(CallbackEvent callbackEvent)
+        {
+            var conferenceId = SourceConference.Id;
+
+            var participantId = callbackEvent.ParticipantId;
+
+            var message = $"Recording Connection Failed: Conference Id: {conferenceId} - Participant id: {participantId}";
+            Logger.LogTrace(
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb.EventHub/Handlers/RecordingConnectionEventHandler.cs:24]:
```suggestion
            Logger.LogTrace("Recording Connection Failed: Conference Id: {ConferenceId} - Participant id: {ParticipantId}", conferenceId, participantId);
```

Structured logging, as the suggested implementation shown, allows log tools to better parse and analyse log data

File: VideoWeb/VideoWeb.UnitTests/EventHandlers/RecordingConnectionEventHandlerTests.cs (hunk 2 above)
Comment (shaed-parkar) [reply VideoWeb/VideoWeb.UnitTests/EventHandlers/RecordingConnectionEventHandlerTests.cs:37]: the ticket requires that the `RecordingConnectionFailed` be invoked

**Review by shaed-parkar — COMMENTED** [review]
Almost there, the PR doesn't meet the ticket requirements

File: VideoWeb/VideoWeb.EventHub/Handlers/RecordingConnectionEventHandler.cs (hunk 3 above)
Comment (marcogagliardi) [reply VideoWeb/VideoWeb.EventHub/Handlers/RecordingConnectionEventHandler.cs:24]: this is sonar suggestion

Comment (shaed-parkar) [reply VideoWeb/VideoWeb.EventHub/Handlers/RecordingConnectionEventHandler.cs:24]: no, this is my suggestion. With all the recent work we need to make sure we use structured logging

Comment (shaed-parkar) [reply VideoWeb/VideoWeb.EventHub/Handlers/RecordingConnectionEventHandler.cs:24]: even if it is a sonar suggestion, it's a good one to implement

Comment (marcogagliardi) [reply VideoWeb/VideoWeb.EventHub/Handlers/RecordingConnectionEventHandler.cs:24]: Sorry I was not clear. The change I did is a Sonar suggestion. Your suggested change is flagged from Sonar as a Maintainability issue.

Comment (marcogagliardi) [reply VideoWeb/VideoWeb.EventHub/Handlers/RecordingConnectionEventHandler.cs]: However I have re-changed the log

Comment (shaed-parkar) [reply VideoWeb/VideoWeb.EventHub/Handlers/RecordingConnectionEventHandler.cs]: Cool I can see the change and Sonar is happy with it :)

**Review by shaed-parkar — APPROVED** [review]

**Review by will-craig — APPROVED** [review]

**Pull request merged**